Stratis.Bitcoin/BlockPulling/LookaheadBlockPuller.cs

```diff
@@ -385,38 +428,46 @@ private Block NextBlockCore(CancellationToken cancellationToken)
                 {
                     if (header.Previous.HashBlock != this.location.HashBlock)
                     {
-                        //A reorg
-                        return null;
+                        this.logger.LogTrace("Blockchain reorganization detected.");
+                        break;
                     }
                     this.IsStalling = false;
                     this.location = header;
                     Interlocked.Add(ref this.currentSize, -block.Length);
                     this.consumed.Set();
-                    return block.Block;
+
+                    res = block.Block;
                 }
                 else
                 {
-                    // Otherwise we either have reorg.
+                    // Otherwise we either have reorg, or we don't know the next block.
                     if (header == null)
                     {
                         if (!this.Chain.Contains(this.location.HashBlock))
                         {
-                            //A reorg
-                            return null;
+                            this.logger.LogTrace("Blockchain reorganization detected.");
+                            break;
                         }
+
+                        this.logger.LogTrace("Hash of the next block is not known.");
                     }
                     else
                     {
+                        this.logger.LogTrace($"Block not available.");
+
                         // Or the block is still being downloaded or we need to ask for this block to be downloaded.
                         if (!isDownloading) AskBlocks(new ChainedBlock[] { header });
 
                         OnStalling(header);
                         this.IsStalling = true;
                     }
-                    WaitHandle.WaitAny(new[] { this.pushed, cancellationToken.WaitHandle }, waitTime[i]);
-                    i = i == waitTime.Length - 1 ? 0 : i + 1;
+
+                    WaitHandle.WaitAny(new[] { this.pushed, cancellationToken.WaitHandle }, WaitNextBlockRoundTimeMs);
```

Comment:
I would suggest we have two types of delays one during IBD and the other if we are at the tip 
(this can be know based on the else of this code ```if (!this.Chain.Contains(this.location.HashBlock))```.

IBD we keep it at 100 ms but if the chain is at the TIP we can potentially wait longer 500 ms (now blocks will arrive in the range a 1-10 min)

Only issue with this might be block propagation, it means a block will be delayed up to 500 ms longer before we ask it validate it and broadcast it.